### ISSUE TITLE
💅 Ambient types appear unused if not used in the same file

### ISSUE THREAD

**Issue by sparr** [biomejs/biome#8351]
### Environment information

```bash
CLI:
  Version:                      2.3.8
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.7.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0

(possibly inaccurate due to use in vscode from global installation)
```

### Rule name

lint/correctness/noUnusedVariables

### Playground link

https://biomejs.dev/playground/?lintRules=noUnusedVariables&files.main.ts=YwBvAG4AcwB0ACAAeAA6AEYAbwBvACAAPQAgADEAOwAKAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoAHgAKQA7AA%3D%3D&files.types.ts=dAB5AHAAZQAgAEYAbwBvAD0AbgB1AG0AYgBlAHIAOwA%3D

### Expected result

No error. The type is ambient/global because types.ts contains no import or export statement, and it is used in another file.

This is probably related to https://github.com/biomejs/biome/issues/8331 which also features a failure to interpret ambient types as global in a different scenario.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
I think this may have been fixed in the mean time. At least I don't see an error in the playground now.

Feel free to reopen if I'm mistaken.
